components/base/sensorcontrolled/velocities.go

```diff
@@ -0,0 +1,135 @@
+package sensorcontrolled
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/control"
+	"go.viam.com/rdk/utils"
+)
+
+type linearVelControllable struct {
+	control.Controllable // is this necessary?
+	sb                   base.Base
+	ms                   movementsensor.MovementSensor
+}
+
+func (l *linearVelControllable) SetState(ctx context.Context, state float64) error {
+	return l.sb.SetVelocity(ctx, r3.Vector{X: 0, Y: state, Z: 0}, r3.Vector{X: 0, Y: 0, Z: 0}, nil)
+}
+
+func (l *linearVelControllable) State(ctx context.Context) (float64, error) {
+	vel, err := l.ms.LinearVelocity(ctx, nil)
+	if err != nil {
+		return math.NaN(), err
+	}
+	return vel.Y, nil
+}
+
+type angularVelControllable struct {
+	control.Controllable
+	sb base.Base
+	ms movementsensor.MovementSensor
+}
+
+func (l *angularVelControllable) SetState(ctx context.Context, state float64) error {
```

Comment:
why not using a single loop ?

Reply:
So here's the architecture I'm trying to implement, since the two endpoints are different - we have an angular velocity sensor and actuator in one endpoint and an linear velocity sensor and actuator in another - both have desired states set by the API and command signals sent by the loop. 
![IMG_2445](https://github.com/viamrobotics/rdk/assets/35934754/39c12596-a8fd-48c7-8707-750abc233ea7)

Reply:
Should MoveStraight and Spin just be considered special cases of SetVelocity (as in they should call SetVelocity with special inputs)? Then you could maybe get away with one control loop but more complicated math? But I suppose this might mean what's in the green square now needs to be handled by the control loop and not the dependent base code?

Reply:
Oh I just realized that only a single state parameter is passable (even though you mentioned it above). My bad

Reply:
I have an idea for how to coordinate the linear and angular velocity states, but I'm having trouble communicating it here. Maybe we could talk offline?

Reply:
> Should MoveStraight and Spin just be considered special cases of SetVelocity (as in they should call SetVelocity with special inputs)? Then you could maybe get away with one control loop but more complicated math? But I suppose this might mean what's in the green square now needs to be handled by the control loop and not the dependent base code?

Yes, not in this pr.

Reply:
> I have an idea for how to coordinate the linear and angular velocity states, but I'm having trouble communicating it here. Maybe we could talk offline?

Yeah, sometime next week.

Reply:
Note: the diagram is no longer valid for the current loop - it is a single loop with velocity input and velocity outputs - no derivatives were used, a simple constant -> sum_block -> gain -> pid -> endpoint -> (back to) sum_block structure is now used. Drawing incoming.